### ISSUE TITLE
python: GLIBC_2.36' not found when python language is enabled

### ISSUE THREAD

**Issue by imincik** [cachix/devenv#819]
**Describe the bug**
When I enable Python language in `devenv.nix` file and try to enter shell environment it fails with following error:
```
/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/bin/python: /nix/store/76l4v99sk83ylfwkz8wmwrm4s8h73rhd-glibc-2.35-224/lib/libc.so.6: version `GLIBC_2.36' not found (required by /nix/store/fcdkcrsrg289h6k33fwmfkhsdj55r6yr-libxcrypt-4.4.36/lib/libcrypt.so.2)
```

**To reproduce**
Please provide an [Short, Self Contained, Correct (Compilable), Example](http://sscce.org) by [creating a gist](https://gist.github.com/) using `devenv.nix`, `devenv.yaml`, and optionally `devenv.lock`.

Make sure to include full logs and what you expected to happen.

* Create following devenv configuration:

devenv.yaml
```
inputs:
  nixpkgs:
    url: github:NixOS/nixpkgs/nixpkgs-unstable
```

devenv.nix
```
{ pkgs, ... }:

{
  languages.python.enable = true;
}
```
* Try to enter devenv shell
```
devenv shell
/nix/store/iw1vmh509hcbby8dbpsaanbri4zsq7dj-python3-3.10.10/bin/python: /nix/store/76l4v99sk83ylfwkz8wmwrm4s8h73rhd-glibc-2.35-224/lib/libc.so.6: version `GLIBC_2.36' not found (required by /nix/store/fcdkcrsrg289h6k33fwmfkhsdj55r6yr-libxcrypt-4.4.36/lib/libcrypt.so.2)
```

**Version**
 
Paste the output of `$ devenv version` here.
```
devenv: 0.6.3
```

**Comment by bobvanderlinden** [comment]
This is probably related to https://github.com/cachix/devenv/issues/749. The problems is that `LD_LIBRARY_PATH` is set to paths in devenvs `packages`. That includes glibc, so any application will be dynamically linked against the glibc from devenv. Many libraries (including libraries that devenv itself indirectly depends on) are not compatible with older/newer glibc versions.

https://github.com/cachix/devenv/pull/745 holds a solution to this problem.

**Comment by imincik** [comment]
Thank you, looking forward for #745

**Comment by mcdonc** [comment]
I think this is the same bug as #555.  I left a comment in that issue at https://github.com/cachix/devenv/issues/555#issuecomment-1885864369 explaining the circumstances that will fix this bug.

**Comment by domenkozar** [comment]
I believe this is fixed in 1.0.1, please reopen if not.
